vms/platformvm/validators/fee/fee.go

```diff
@@ -91,77 +91,46 @@ func (s State) CostOf(c Config, seconds uint64) uint64 {
 	return cost
 }
 
-// SecondsUntil calculates the number of seconds that it would take to charge at
-// least [targetCost] based on the dynamic fee mechanism. The result is capped
-// at [maxSeconds].
-func (s State) SecondsUntil(c Config, maxSeconds uint64, targetCost uint64) uint64 {
+// SecondsRemaining calculates the maximum number of seconds that a validator
+// can pay fees before their [costLimit] would be exceeded based on the dynamic
+// fee mechanism. The result is capped at [maxSeconds].
+func (s State) SecondsRemaining(c Config, maxSeconds uint64, costLimit uint64) uint64 {
 	// Because this function can divide by prices, we need to sanity check the
 	// parameters to avoid division by 0.
 	if c.MinPrice == 0 {
-		if targetCost == 0 {
-			return 0
-		}
 		return maxSeconds
 	}
 
 	// If the current and target are the same, the price is constant.
 	if s.Current == c.Target {
-		price := gas.CalculatePrice(c.MinPrice, s.Excess, c.ExcessConversionConstant)
-		return secondsUntil(
-			uint64(price),
-			maxSeconds,
-			targetCost,
-		)
+		price := uint64(gas.CalculatePrice(c.MinPrice, s.Excess, c.ExcessConversionConstant))
+		seconds := costLimit / price
+		return min(seconds, maxSeconds)
 	}
 
-	var (
-		cost    uint64
-		seconds uint64
-		err     error
-	)
-	for cost < targetCost && seconds < maxSeconds {
+	for seconds := uint64(0); seconds < maxSeconds; seconds++ {
 		s = s.AdvanceTime(c.Target, 1)
 
 		// Advancing the time is going to either hold excess constant,
 		// monotonically increase it, or monotonically decrease it. If it is
 		// equal to 0 after performing one of these operations, it is guaranteed
 		// to always remain 0.
 		if s.Excess == 0 {
-			zeroExcessCost := targetCost - cost
-			secondsWithZeroExcess := secondsUntil(
-				uint64(c.MinPrice),
-				maxSeconds,
-				zeroExcessCost,
-			)
-
+			secondsWithZeroExcess := costLimit / uint64(c.MinPrice)
 			totalSeconds, err := safemath.Add(seconds, secondsWithZeroExcess)
-			if err != nil || totalSeconds >= maxSeconds {
+			if err != nil {
+				// This is technically unreachable, but makes the code more
+				// clearly correct.
 				return maxSeconds
```

Comment:
As above re `0` instead of `maxSeconds`

Reply:
If this value were to overflow, it would mean that `maxSeconds` should be retuned. `0` would not be correct to return (if this value weren't well defined, we'd need to return an error rather than an ambiguous `0`)

Reply:
Similar clarification to above - this function is essentially implementing:

"If you need to pay $x every second, how long does $y last? If the answer is more than z then just say 1 minute."

If $x would last longer than MaxUint64, returning 0 is incorrect. We should return 1 minute.